odak/learn/lensless/camera.py

```diff
@@ -0,0 +1,39 @@
+import torch.fft as fft
+
+from ..tools import crop_center, zero_pad
+
+
+class LenslessCamera:
+    def __init__(self, *args, **kwargs):
+        raise NotImplementedError
+
+
+class PhaseMaskCamera(LenslessCamera):
+    """
+    Forward and adjoint models for a phase-mask based lensless camera with a large, shift invariant kernel.
+    The fourier transformed point spread function (PSF) is cached for efficiency.
+
+    Parameters
+    ----------
+    psf         : torch.tensor
+                  Real-valued point spread function measurement.
+    """
+    def __init__(self, psf):
+        self.h = rft(zero_pad(psf))
+
+    def autocorrelation(self):
+        return self.h * self.h.conj()
+
+    def forward(self, scene):
+        return crop_center(irft(self.h * rft(scene)))
+
+    def adjoint(self, error):
+        return irft(self.h.conj() * rft(zero_pad(error)))
+
+
+def rft(x):
+    return fft.rfft2(fft.ifftshift(x))
```

Comment:
I'd like to use `odak.learn.tools.convolve2d` here, but we would need to agree on conventions for fourier transforms, which is a bit trickier.

- It's helpful to always shift the kernel and the image such that they always occupy the same region in fourier space
- By storing the output of a fourier transform in `h`, we can compute the adjoint by flipping the conjugate bit (`h.conj()` is a single bit flip)
- When working with real valued point spread functions, we can use `rfft` which offers a 2x performance boost

For now I propose leaving these functions as is, but it's probably worth establishing a convention for fourier transforms in odak at some point